projects/wave-core-new/src/lib/time/time-slider/time-slider.component.ts

```diff
@@ -0,0 +1,193 @@
+        private changeDetectorRef: ChangeDetectorRef,
+    ) {
+        this.subscriptions.push(
+            this.projectService.getLayerStream().subscribe((layerList) => {
+                if (layerList !== this.layerList) {
+                    this.layerList = layerList;
+                }
+                this.changeDetectorRef.markForCheck();
+            }),
+        );
+    }
+
+    ngOnInit(): void {
+        this.getTimelineData();
+        this.getTimelineGroups();
+        this.getOptions();
+
+        this.timeline = new Timeline(this.timelineContainer.nativeElement, this.data, this.options);
+        this.timeline.setGroups(this.groups);
+        this.timeline.setItems(this.data);
+
+        this.timeline.addCustomTime(this.startTime, 'start');
+        this.timeline.addCustomTime(this.endTime, 'end');
+
+        //listens to events of the custom timebars
+        this.timeline.on('timechanged', (properties) => {
+            if (properties.id === 'start') {
+                this.startTime = this.timeline?.getCustomTime(properties.id) as Date;
+            }
+            if (properties.id === 'end') {
+                this.endTime = this.timeline?.getCustomTime(properties.id) as Date;
+            }
+            if (!this.isRange) {
+                this.endTime = this.startTime;
+            }
+            this.changeTime();
+        });
+
+        this.timeline.on('rangechanged', (properties) => {
+            const startWindow = properties.start;
+            const endWindow = properties.end;
+            const windowSize = endWindow - startWindow;
+            this.changeSelectedScale(windowSize);
+        });
+
+        //Connection to the timestream of the project
+        //changes the position of the Timebars according to the Timestream of the project
+        this.subscriptions.push(
+            this.projectService.getTimeStream().subscribe((t) => {
+                this.startTime = t.start.toDate();
+                this.endTime = t.end.toDate();
+                this.changeDetectorRef.markForCheck();
+                if (this.isRange) {
+                    this.timeline?.removeCustomTime('end');
+                    this.isRange = false;
+                }
+                if (t.end.isAfter(t.start)) {
+                    this.timeline?.addCustomTime(t.end.toDate(), 'end');
+                    this.isRange = true;
+                }
+                this.timeline?.setCustomTime(t.start.toDate(), 'start');
+            }),
+        );
+    }
+
+    changeSelectedScale(windowSize: number): void {
+        if (windowSize > 1000 * 60 * 60 * 24 * 31 * 22) this.selectedScale = 'year';
+        else if (windowSize > 1000 * 60 * 60 * 24 * 15) this.selectedScale = 'month';
+        else if (windowSize > 1000 * 60 * 60 * 31) this.selectedScale = 'day';
+        else this.selectedScale = 'hour';
+    }
+
+    //changes the Timestream of the project according to the Timebars
+    changeTime(): void {
```

Comment:
i think you need to check if the time is local or UTC time. When the time is set to 01.01.2014 00:00 and you use the slider to move it to the next month it will change to 31.01.2014 23:00

Reply:
you were right... I had to switch it to UTC
1b7e67ec0611b20d5198bd6bb46d909aab95d83a